delphi-frontend/src/main/java/au/com/integradev/delphi/cfg/api/Block.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2024 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.cfg.api;
+
+import java.util.List;
+import java.util.Set;
+import org.sonar.plugins.communitydelphi.api.ast.DelphiNode;
+
+public interface Block {
+  /** Unique block identifier */
+  int getId();
```

Comment:
Something we've pretty directly ripped off from sonar-java.
The unique identifier is all well and good - but what's it actually for?

The only usage in application code seems to be ordering a `PriorityQueue` based on them (?) which I'd have questions about. Are we relying on the implementation detail of a linear ordering based on creation order for this "unique block identifier", or am I misunderstanding something?

Reply:
After discussion, the notion of the ID will remain. However, it will not be in the API and predominantly used for debugging.